### PR TITLE
[89] - list_with_label_selector namespace as a ref + tests

### PR DESCRIPTION
**Pull request by Pscheidl**
Originally added namespaces support to `list_with_label_selector` method. This has been implemented in the meantime already and merged to main. Therefore, this PR adds the following:

1. Namespaces are passed as string slices, therefore no strings are moved or even cloned anymore.
2. Tests

### CODE REVIEW

**Comment by lfrancke** [comment]
Thank you! 
Let's wait for #78 to be merged as I believe this already adds the namespace stuff (but not the other changes).

**Comment by Pscheidl** [comment]
Not ready to merge yet, as I'll have to do some adjustments manually.

**Comment by Pscheidl** [comment]
@lfrancke Updated PR description. Ready to merge, provided you like it. It adds tests and makes sure only a `usize` ptr is cloned and moved around, no the clone of the whole `String`.

**Review by lfrancke — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**
